### PR TITLE
Added unit tests for EmptyDirectionsGuard, RequestDispatcher, RequestQueue

### PR DESCRIPTION
**Pull request by felixlapierre**
Added unit tests for the following classes: EmptyDirectionsGuard, RequestDispatcher, RequestQueue

Some changes were made to accomodate / supplement these tests
- Reworked how error message popups worked in order to remove dependencies on Activity
- Added test suites for unit & acceptance tests

There are a couple of empty interfaces in this pull request but they'll be used in my next one

Current test coverage:
image

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/directions/Directions.kt [hunk 1]
```diff
@@ -31,17 +37,12 @@ class Directions constructor(private var activity: AppCompatActivity){
         val request = JsonObjectRequest(Request.Method.GET, url, null,
```
Comment (FGRCL) [inline app/src/main/java/com/example/campusguide/directions/Directions.kt:37]: Why not use your VolleyRequest Class here?

File: app/src/main/java/com/example/campusguide/directions/EmptyDirectionsGuard.kt [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 
-import android.app.Activity
-import android.app.AlertDialog
-import androidx.appcompat.app.AppCompatActivity
-import com.example.campusguide.utils.MessageDialogFragment
+import com.example.campusguide.utils.ErrorListener
 
-class EmptyDirectionsGuard constructor(private val activity: AppCompatActivity, private val wrapped: GetDirectionsDialogFragment.DirectionsDialogConfirmationListener)
-    : GetDirectionsDialogFragment.DirectionsDialogConfirmationListener {
+class EmptyDirectionsGuard constructor(private val wrapped: DirectionsDialogConfirmationListener, private val errorListener: ErrorListener)
```
Comment (FGRCL) [inline app/src/main/java/com/example/campusguide/directions/EmptyDirectionsGuard.kt:5]: Is this input validation for the directions?

**Review by FGRCL — APPROVED** [review]

File: app/src/main/java/com/example/campusguide/directions/Directions.kt (hunk 1 above)
Comment (felixlapierre) [reply app/src/main/java/com/example/campusguide/directions/Directions.kt:37]: Will be used in the next PR

File: app/src/main/java/com/example/campusguide/directions/EmptyDirectionsGuard.kt (hunk 2 above)
Comment (felixlapierre) [reply app/src/main/java/com/example/campusguide/directions/EmptyDirectionsGuard.kt:5]: Yep, this opens a simple message box if the user leaves start or end blank

**Pull request merged**